### ISSUE TITLE
Advice on cocurrent asyncio testing

### ISSUE THREAD

**Issue by dimaqq** [pytest-dev/pytest#4526]
Hi pytest folks.
[obligatory bow for amazing and essential testing framework, library and runner]

I'm toying with the idea of running asyncio tests concurrently while maintaining fixture expectations.

There's `pytest-concurrent` that promises same for threads and gevent, but not asyncio.
I'm fully aware that only a small subset of tests could be ran concurrently, specifically mocking is out of scope.
I'd use this for functional black-box tests.

Are there other plugins I should look at first?
Where can I hook into fixture life cycle?
What about asyncio checks after test is over (e.g. future not awaited)?
If it generally a bad idea for some other reason?

**Comment by blueyed** [comment]
I suggest looking into pytest-xdist if it's about running tests in parallel - you can multiple local cores, or even remote hosts (via ssh, very nice if you have e.g. EC2 instances available anyway - but not trivial to setup/deploy, but Docker helps there).

Apart from that I've used the following (basically running an event loop in
a test itself), but only because I needed to create a race condition:

```python
def test_foo_race(…, live_server, event_loop):
    import asyncio
    import aiohttp
    …
    active_requests_mock.post(
        'https://api.stripe.com/v1/ephemeral_keys', json=key_response)

    url = '%s%s' % (live_server.url, reverse_query(
        'app:foo',
        query_kwargs=…,
        kwargs=…,
    ))

    event_loop.set_debug(True)

    async def get_ephemeralkey():
        async with aiohttp.ClientSession(headers={
            'Accept': 'application/json; version=1.0',
            'Authorization': 'JWT %s' % jwt
        }) as client:
            async with client.get(url) as resp:
                text = await resp.text()
                assert resp.status == 200, text
                assert json.loads(text) == key_response

    coros = asyncio.gather(
        get_ephemeralkey(),
        get_ephemeralkey(),
        loop=event_loop,
    )
    event_loop.run_until_complete(coros)
```

**Comment by Zac-HD** [comment]
[`pytest-asyncio`](https://pypi.org/project/pytest-asyncio/) is the plugin you should be looking at, though pytest-dev/pytest-asyncio#69 suggests that running tests concurrently is not and will not be an option so you might need to write something custom.
